scripts/experiments_tabular/pidforest.jl

```diff
@@ -17,16 +17,27 @@ s = ArgParseSettings()
         help = "dataset"
 end
 parsed_args = parse_args(ARGS, s)
-@unpack dataset, seed = parsed_args
+@unpack dataset, max_seed = parsed_args
 
 modelname = "pidforest"
 function sample_params()
-	par_vec = (6:2:10, 50:25:200, 50:50:200, 3:6, [0.05, 0.1, 0.2], )
+	par_vec = (6:2:10, 50:25:200, [50, 100, 250, 500, 1000, 5000], 3:6, [0.05, 0.1, 0.2], )
 	argnames = (:max_depth, :n_trees, :max_samples, :max_buckets, :epsilon, )
 
 	return (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
 end
 
+function GenerativeAD.edit_params(data, parameters)
+	D, N = size(data[1][1])
+	if N < parameters.max_samples
+		# if there are not enough samples, choose closest multiple of 50
+		@info "Not enough samples in training, changing max_samples for each tree."
+		return merge(parameters, (;max_samples = max(25, N - mod(N, 50))))
```

Comment:
Is the max_samples parameter going to have a discontinuity at N? Since for N >= max_samples it stays the same, but for N = max_samples - 1 the modulo is subtracted. Might that be a problem?

Reply:
My thinking was that:
1. if the dataset has less than 50 samples -> choose for each forest 25 samples
2. if the dataset has more than 50 samples but less than supplied max_samples -> choose some closest multiples of 50
The latter should take care of datasets ranging from 50 to hundreds. For bigger datasets there are usually many valid choices of this hyperparameter, so that editting won't take place that often.